ext/intl/formatter/formatter_format.c

```diff
@@ -53,10 +64,23 @@ PHP_FUNCTION( numfmt_format )
 			case IS_DOUBLE:
 				type = FORMAT_TYPE_DOUBLE;
 				break;
-			EMPTY_SWITCH_DEFAULT_CASE();
+			case IS_STRING:
+				type = FORMAT_TYPE_DECIMAL;
+				break;
+			default:
+				zend_argument_type_error(1, "must be of type int|float|string, %s given", zend_zval_type_name(number));
```

Comment:
with `Z_PARAM_STR_OR_NUMBER`, can this line be reached?

Reply:
Hm, good point, I think it can’t be – at least I can’t seem to reach it with some manual testing. (All the error messages I get name the parameter type as `string|int|float`, not `int|float|string` as in this line, so those errors presumably come from `Z_PARAM_STR_OR_NUMBER` instead.)

I seem to have introduced this in #9909, possibly misunderstanding [this comment](https://github.com/php/php-src/pull/7782#discussion_r886547876) despite otherwise not going back to `zend_parse_method_parameters()` after all. I’ll change it back and see if any CI complains.

Reply:
I do think the approach highlighted by @cmb69 to use a generic ``z`` to be potentially more sensible.

Reply:
Done. (I think I misunderstood the suggestion before.) Makes the commit smaller, at least, which is nice.